### DISCUSSION TITLE
Printing multiple tabs at the same time

### DISCUSSION THREAD

**Discussion by Sergio3N** [sumatrapdfreader/sumatrapdf#4015] [Ideas]
Hi everyone, I'm a new user of the program. I wanted to ask if it is possible to implement the **simultaneous printing of multiple tabs** at the same time, selecting them with CTRL or SHIFT.

None of the PDF viewing programs have this very convenient option, it often happens during the working day that I open multiple PDFs at the same time and have to print them all...

**Why not be the first to do it?** 😜

Sorry for my English

**Comment by GitHubRulesOK** [comment]
there is a reason that from the days before CP/M (the system DOS was built upon, and then later MS windows) does not print multiple files is because you will likely get a runaway system, Traditionally that means not being able to open the print room door because all the paper was spewed out at once overnight.

Joking apart it is never a good idea to send multiple unMANaged prints. So printing is decidedly a human task, best not done by androids nor robots, and for that reason no browsing app is generally designed for servers to print more than one document at a time.

**Comment by Sergio3N** [comment]
😥

**Comment by Dobrowit** [comment]
If sending many files to the print queue at once is not a good idea, there is another way to accomplish this task - more responsible (to the environment). **Merging open files into one.** For example, in the order in which the tabs are arranged from left to right. The glued file would appear as the last tab and could be printed as one document sent to the queue.

I work in a place where people print documents on a daily basis. I often receive **many (a dozen or so) PDF files in a package and printing them one by one in SumatraPDF is a pain.**

First of all, SumatraPDF does not remember printer settings (even during one session with several tabs). If I have 15 PDF files open (today, for example, I had such a task), I have to go into the printer settings each time to select the appropriate one (I have several printers), set the format, orientation, etc. In such a case, it is easy to make a mistake (and this happened today - I had to repeat two printouts because they were printed on A4 and were supposed to be on A3). It is better to perform such a simple task as printing many open documents in Adobe Reader (which I hate :-)) - this program remembers the first printer setting and when printing subsequent documents, it remembers how the printer was set and what parameters it had set.

Secondly, it would be useful to detect and remove blank pages. Even a function to manually delete selected pages would be very useful. Sometimes I receive scans of document files that were scanned on both sides just because one or two pages were printed on both sides.

Thirdly, the above functions would be complemented by the ability to insert blank pages and the ability to rotate selected pages.

**Reply by GitHubRulesOK** [reply]
@Dobrowit 

Ok lets try to address some points objectively

SumatraPDF  Is a fantastic PDF (and other formats) viewer 
and I have devoted a lot of time to explaining its benefits, however it is NOT ideal as a PDF print manager.
(Note as NOT an editor then permanent rotating or removing pages  are not its remit but could be done by calling a CLI)


When you combine files together there are disadvantages and advantages, so in this case the number of pages will often lead to overcomplex resource usage and potential increased memory overheads.

I am not discounting the idea of accumulate tabs into one but it is likely to cause more issues for many users.

On the basis you wish to store a set of instructions to be batch run it may be better to export a list of desires, 
thus tab 1 to send pages 1 and 6 as A3 landscape to plotter but tab 2 to send whole file to small printer as A4.

Now the new problem is that's a lot of specialised code for a fairly simple batch driven list. and currently only possible by echo the current tab name and current page number into a CMD file to make a choice such as page ranges like 1,3,5-6 starts to require lots of checking but push current file name to a list for 1 of 2 printers is easy and it could be prefixed by a CMD `set /p =question` of which pages are to be included.

So the day to day niggles could be potentially reduced by adding "ExternalViewers" output  into a batch file to run at the end of a series of questions.
